### PR TITLE
Port ValidateFormatString analyzer and tests to shared layer

### PR DESCRIPTION
**Pull request by mavasani**
**NOTE:** The option used by this analyzer is not an editorconfig based option, so it has no effect on CI execution in CodeStyle layer. I have modified the options tests to account for this behavior.

First commit: File moves + options related test changes
Second commit: xlf and resx changes

### CODE REVIEW

File: src/Analyzers/CSharp/Tests/ValidateFormatString/ValidateFormatStringTests.cs [hunk 1]
```diff
@@ -544,33 +537,35 @@ static void Main(string[] args)
 ");
         }
 
-        [Fact, Trait(Traits.Feature, Traits.Features.ValidateFormatString)]
-        public async Task WarningTurnedOff()
```
Comment (mavasani) [inline src/Analyzers/CSharp/Tests/ValidateFormatString/ValidateFormatStringTests.cs]: `WarningTurnedOff` was broken in both the C# and VB tests. I have combined it with `WarningTurnedOn` to execute on the same source but with different `expectDiagnostic` values.

File: src/Analyzers/Core/Analyzers/ValidateFormatString/ValidateFormatStringOption.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.ValidateFormatString
+{
+    internal class ValidateFormatStringOption
```
Comment (mavasani) [inline src/Analyzers/Core/Analyzers/ValidateFormatString/ValidateFormatStringOption.cs]: Split into a separate file from `ValidateFormatStringOptionProvider`, latter is only defined in Workspaces layer.

File: src/Analyzers/VisualBasic/Tests/ValidateFormatString/ValidateFormatStringTests.vb [hunk 3]
```diff
@@ -18,24 +19,18 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.ValidateFormatStri
...
-                {New OptionKey2(ValidateFormatStringOption.ReportInvalidPlaceholdersInStringDotFormatCalls, LanguageNames.CSharp), True},
-                {New OptionKey2(ValidateFormatStringOption.ReportInvalidPlaceholdersInStringDotFormatCalls, LanguageNames.VisualBasic), False}
-            }
-
-            Return optionsSet
+        Private Function VBOptionOffCSharpOptionOn() As IOptionsCollection
+            Return OptionsSet(
+                (New OptionKey2(ValidateFormatStringOption.ReportInvalidPlaceholdersInStringDotFormatCalls, LanguageNames.CSharp), True),
+                (New OptionKey2(ValidateFormatStringOption.ReportInvalidPlaceholdersInStringDotFormatCalls, LanguageNames.VisualBasic), False)
+            )
```
Comment (CyrusNajmabadi) [inline src/Analyzers/VisualBasic/Tests/ValidateFormatString/ValidateFormatStringTests.vb]: nit: we try to avoid naked close parens in IDE.

Comment (mavasani) [reply src/Analyzers/VisualBasic/Tests/ValidateFormatString/ValidateFormatStringTests.vb]: Ah sure, I will fix it given I already have merge conflicts. Just curious - do we have any CodeStyle analyzer that will flag this?

**Pull request merged**